include/libtorrent/aux_/io.hpp

```diff
@@ -59,12 +59,16 @@ namespace libtorrent { namespace aux
 		return ret;
 	}
 
-	template <class T, class Byte>
+	template <class T, class In, class Byte, typename Cond
+		= typename std::enable_if<std::is_integral<In>::value
```

Comment:
`sizeof(In) == 1`? The type `In` is the one we don't want to cast, the one that is restricted already in the `enable_if`.

I will add the condition to `T`.